Makefile

```diff
@@ -522,6 +523,7 @@ update-dependencies:
 
 go-generate:
 	@printf $(COLOR) "Process go:generate directives..."
+	@go build -o cached-mockgen ./cmd/tools/mockgen
```

Comment:
I'm not sure I like this cluttering up the root of the checkout.. why not `go install` it like other build tools? (name the package something other than `mockgen` so it doesn't interfere on PATH)

Reply:
Done.